test.functional/build.gradle.kts

```diff
@@ -1,7 +1,62 @@
+import com.moowork.gradle.node.yarn.YarnTask
+import java.util.function.Consumer
+
 plugins {
     id("java")
+    id("com.cognifide.aem.instance")
+    id("com.moowork.node")
 }
 
 description = "Example - Functional Tests"
 
+tasks{
+    register<YarnTask>("testFunctional") {
+        dependsOn("testSetup", "testInstallNode")
+        group = "check"
+        setYarnCommand("jest")
+        setWorkingDir(file("${projectDir.absolutePath}/test.functional"))
+        outputs.dir(file("${projectDir.absolutePath}/test.functional"))
+        doFirst {
+            val props = mutableListOf()
+            if (project.hasProperty("aem.env")) {
+                props.addAll(listOf("--config", "./env/${project.properties["aem.env"]}.config.js"))
+            } else {
+                props.addAll(listOf("--config", "./env/local-publish.config.js"))
+            }
+            args = props
+        }
+    }
+
+    register("testSetup") {
+        description = "Creates env json from gradle properties."
+        group = "check"
+        doLast {
+            val configTemplate = File("${projectDir.absolutePath}/config/_templates/template-jest-config.js").readText()
+            val envTemplate = File("${projectDir.absolutePath}/config/_templates/template-puppeteer-environment.js").readText()
+            file("${projectDir.absolutePath}/env").mkdirs()
+            aem.instances.forEach(Consumer {
+                val values = hashMapOf("instanceName" to it.name, "instanceUrl" to it.httpUrl)
```

Comment:
whole `instance` is serializable, so maybe add default method  `Instance.toJson` in interface and use it here to pass all details to generate file

Reply:
Done